### PR TITLE
fixes #131

### LINKED ISSUE #131: Allow login with email and password
Currently only login with google is possible. There will be people without a google account and instead have proton mail accounts or some other email provider. So it would be nice to have that option

### PR DESCRIPTION
**Pull request by Yh010**
# Add Sign-in with Email and Logout Features

## Overview
This pull request introduces a new email-based sign-in feature and a logout functionality to enhance user authentication in the application. It includes updates to the authentication flow, UI components, and header navigation to provide a seamless user experience.

## Changes
- New Features:
  - Added an `EmailSignInForm` component that allows users to sign in using their email address.
  - Implemented a `SignOut` component in the header to enable users to securely log out of the application.
  - Integrated the `signIn` function from `next-auth/react` to handle email-based authentication flow.
  
- Refactoring:
  - Created a reusable `TextField` component to manage form input fields more efficiently.
  - Improved the overall structure and organization of authentication-related files and components for better maintainability.

- Other Changes:
  - Updated the `package.json` to include new dependencies: `@react-email/components`, `nodemailer`, and `resend`.
  - Modified database configuration in `wrangler.toml` to point to new database names and IDs.
  - Enhanced the email sign-in functionality to send a magic link for authentication.
  - Adjusted imports in `auth.ts` to include the new utility function for sending verification emails.

> ✨ Generated with love by Kaizen ❤️

<details>
<summary>Original Description</summary>
Loom:
https://www.loom.com/share/58c24c5c9cbf49a1907e55b35fa2f23d?sid=ae736d7b-7f4a-4545-9a03-33185745f8db

This PR adds:
- signin with email feature
- logout feature
</details>

### CODE REVIEW

File: apps/web/wrangler.toml [hunk 1]
```diff
@@ -26,22 +26,22 @@ bucket_name = "dev-r2-anycontext"
 
 [[d1_databases]]
 binding = "DATABASE"
-database_name = "dev-d1-anycontext"
-database_id = "fc562605-157a-4f60-b439-2a24ffed5b4c"
+database_name = "supermemory-db-preview"
```
Comment (CodeTorso) [inline apps/web/wrangler.toml:29]: for local development only this database is required, you do not need to create the other two databases

**Comment by CodeTorso** [comment]
great! looked into the video, we can make it work

but it uses resend, there free tier is 3k per month, considering the fact that supermemory grew to 6.5k users in a month, I fear we will cross that limit. 

@Dhravya your thoughts ?

also you have pushed wrangler.toml with different values, It breaks prodution.

**Comment by Yh010** [comment]
> uses resend, there free tier is 3k per month, considering the fact that supermemory grew to 6.5k users in a month, I fear we will cross that limit.

Makes sense. I'll see what alternative I can find

**Comment by Yh010** [comment]
> have pushed wrangler.toml with different values, It breaks prodution.

I know man, accidentally pushed it and now I don't know how to remove that🥲🥲🥲

Actually my laptop's hard disk has crashed, so I won't be able to work for at least 2-3 days.

**Comment by CodeTorso** [comment]
> > have pushed wrangler.toml with different values, It breaks prodution.
> 
> I know man, accidentally pushed it and now I don't know how to remove that🥲🥲🥲
> 
> Actually my laptop's hard disk has crashed, so I won't be able to work for at least 2-3 days.

It's fine.

**Comment by Dhravya** [comment]
i'll have to heavily test this with other stuff.
right now I think we'll have the google only because of how the extension and mobile auth work

**Comment by CodeTorso** [comment]
> i'll have to heavily test this with other stuff. right now I think we'll have the google only because of how the extension and mobile auth work

yeah, I was thinking of it too. When I checked @aryasaatvik's github signin pr

**Comment by Dhravya** [comment]
After reviewing i found that this breaks a bunch of stuff right now
For eg. The chrome extension doesn't work with the watch

Soon we'll move to giving API keys to each user and we'll open this again when that's ready

**Pull request closed without merging**